tarantool_test.go

```diff
@@ -693,6 +695,484 @@ func TestClient(t *testing.T) {
+				Data:     []interface{}{},
+				MetaData: SQLMetaData{},
+			},
+		},
+	}
+
+	var conn *Connection
+	conn, err = Connect(server, opts)
+	assert.NilError(t, err, "Failed to Connect")
+	assert.Check(t, conn != nil, "conn is nil after Connect")
+	defer conn.Close()
+
+	for i, test := range testCases {
+		resp, err := conn.Execute(test.Query, test.Args)
+		assert.NilError(t, err, "Failed to Execute, Query number: %d", i)
+		assert.Check(t, resp != nil, "Response is nil after Execute\nQuery number: %d", i)
+		for j := range resp.Data {
+			assert.DeepEqual(t, resp.Data[j], test.Resp.Data[j])
+		}
+		assert.Equal(t, resp.SQLInfo.AffectedCount, test.Resp.SQLInfo.AffectedCount)
+
+		for j := range resp.MetaData.ColumnsInfo {
+			assert.Equal(t, resp.MetaData.ColumnsInfo[j].FieldIsAutoincrement, test.Resp.MetaData.ColumnsInfo[j].FieldIsAutoincrement)
+			assert.Equal(t, resp.MetaData.ColumnsInfo[j].FieldIsNullable, test.Resp.MetaData.ColumnsInfo[j].FieldIsNullable)
+			assert.DeepEqual(t, resp.MetaData.ColumnsInfo[j].FieldCollation, test.Resp.MetaData.ColumnsInfo[j].FieldCollation)
+			assert.DeepEqual(t, resp.MetaData.ColumnsInfo[j].FieldName, test.Resp.MetaData.ColumnsInfo[j].FieldName)
+			assert.DeepEqual(t, resp.MetaData.ColumnsInfo[j].FieldSpan, test.Resp.MetaData.ColumnsInfo[j].FieldSpan)
+			assert.DeepEqual(t, resp.MetaData.ColumnsInfo[j].FieldType, test.Resp.MetaData.ColumnsInfo[j].FieldType)
+		}
+	}
+}
+
+func TestSQLBindings(t *testing.T) {
+	// Data for test table
+	testData := map[int]string{
+		1: "test",
+	}
+
+	// Check for skip SQL tests if tarantool version < 2.0.0
+	isLess, err := test_helpers.IsTarantoolVersionLess(2, 0, 0)
+	if err != nil {
+		t.Fatal("Could not check the Tarantool version")
+		return
+	}
+	if isLess {
+		t.Skip()
+	}
+
+	var resp *Response
+	var conn *Connection
+
+	conn, err = Connect(server, opts)
+	if err != nil {
+		t.Errorf("Failed to connect: %s", err.Error())
+		return
+	}
+	if conn == nil {
+		t.Errorf("conn is nil after Connect")
+		return
+	}
+	defer conn.Close()
+
+	resp, err = conn.Execute(createTableQuery, []interface{}{})
+	if err != nil {
+		t.Fatalf("Failed to Execute: %s", err.Error())
+	}
+	if resp == nil {
+		t.Errorf("Response is nil after Execute")
+	}
+	if resp.Code != 0 {
+		t.Errorf("Failed to Execute: %d", resp.Code)
+	}
+	if resp.SQLInfo.AffectedCount != 1 {
+		t.Errorf("Incorrect count of created spaces: %d", resp.SQLInfo.AffectedCount)
+	}
+
+	resp, err = conn.Execute(insertQuery, []interface{}{1, testData[1]})
+	if err != nil {
+		t.Errorf("Failed to Execute: %s", err.Error())
+	}
+	if resp == nil {
+		t.Errorf("Response is nil after Execute")
+	}
+	if resp.Code != 0 {
+		t.Errorf("Failed to Execute: %d", resp.Code)
+	}
+	if resp.SQLInfo.AffectedCount != 1 {
+		t.Errorf("Incorrect count of affected rows: %d", resp.SQLInfo.AffectedCount)
+	}
+
+	// test all types of supported bindings
+	// prepare named sql bind
+	sqlBind := map[string]interface{}{
```

Comment:
Now the only way to understand possible sql binds is to read this test. And now it doesn't look like a good demonstration material. I don't propose to rework current test: it's good as it is. But it would be really nice to add SQL examples to README. I don't know about current plans, so it could be considered excessive for now. But it would be  hard how to use SQL without README example.

Reply:
I have added some examples of using SQL in connector, but we still need to determine some moments. For example, if the set of proposed types of query arguments is not redundant (that especially relates to positioned arguments).

Reply:
After https://github.com/tarantool/go-tarantool/pull/151 we've moved all examples (except for the basic one) to separate example files rather than README.

Reply:
Oh, I see you already made it in https://github.com/tarantool/go-tarantool/pull/144#discussion_r842075016